sessionctx/variable/tidb_vars.go

```diff
@@ -810,6 +811,10 @@ const (
 	TiDBDDLEnableFastReorg = "tidb_ddl_enable_fast_reorg"
 	// TiDBDDLDiskQuota used to set disk quota for lightning add index.
 	TiDBDDLDiskQuota = "tidb_ddl_disk_quota"
+	// TiDBOptRangeMaxSize is the max memory limit for ranges. When the optimizer estimates that the memory usage of complete
+	// ranges would exceed the limit, it chooses less accurate ranges such as full range. 0 indicates that there is no memory
+	// limit for ranges.
+	TiDBOptRangeMaxSize = "tidb_opt_range_max_size"
```

Comment:
It seems wrong here. You can see line747 `// TiDB vars that have only global scope`.

Reply:
OK. I will put it in the right place.

Reply:
#37835 fixes the problem.